confluent_kafka/schema_registry/schema_registry_client.py

```diff
@@ -0,0 +1,588 @@
+            - version (int): version number
+
+        Returns
+            Version: Version of the deleted schema
+
+        Raises:
+            SchemaRegistryClientError if the subject or version cannot be found.
+
+        .. _Schema Registry API Reference:
+            https://docs.confluent.io/current/schema-registry/develop/api.html#delete--subjects-(string-%20subject)-versions-(versionId-%20version)
+
+        """  # noqa: E501
+        self._rest_client.DELETE(url_formatter(VERSION,
+                                               subject_name,
+                                               version))
+
+    def set_compatibility(self, subject_name=None, level=None):
+        """
+        PUT /config/{subject_name}
+
+        Update global or subject level compatibility policy.
+
+        Args:
+            - level (CompatibilityType): Compatibility policy
+
+        Keywork Args:
+            - subject_name (str, optional): Subject name.
+                Sets global policy if not set.
+            - level  (CompatibilityType): Compatibility policy
+        Raises:
+            SchemaRegistryClientError: if the compatibility level is invalid
+
+        .. _Schema Registry API Reference:
+            https://docs.confluent.io/current/schema-registry/develop/api.html#put--config
+
+        """  # noqa: E501
+
+        if level is None:
+            raise ValueError("level must be set")
+
+        print(subject_name)
+        if subject_name is None:
+            return self._rest_client.PUT(url_formatter(CONFIGS),
+                                         body=level,
+                                         encoder=_CompatibilityJSONEncoder,
+                                         decoder=_CompatibilityJSONDecoder)
+
+        return self._rest_client.PUT(url_formatter(CONFIG,
+                                                   subject_name),
+                                     body=level,
+                                     encoder=_CompatibilityJSONEncoder,
+                                     decoder=_CompatibilityJSONDecoder)
+
+    def get_compatibility(self, subject_name=None):
+        """
+        GET /config
+
+        Get the current global Compatibility level.
+
+        Args:
+            - subject_name (str, optional): Subject name. Returns global policy
+                if left unset.
+
+        Returns:
+            str: Compatibility level for the subject if set, otherwise the
+                global compatibility level.
+
+        Raises:
+            SchemaRegistryClientError: if the request was unsuccessful.
+
+        .. _Schema Registry API Reference:
+            https://docs.confluent.io/current/schema-registry/develop/api.html#get--config
+
+        """  # noqa: E501
+        if subject_name is not None:
+            url = url_formatter(CONFIG, subject_name)
+        else:
+            url = url_formatter(CONFIGS)
+
+        return self._rest_client.GET(url, decoder=_CompatibilityJSONDecoder)
```

Comment:
I think I said this before, but I feel you get very little by putting the `.json(cls=decoder, **kwargs)` in the `GET` method and it comes at the cost of extra complexity which makes things harder to follow.